extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -20,35 +20,157 @@
 // Most other ad networks will want to put their A4A code entirely in the
 // extensions/amp-ad-network-${NETWORK_NAME}-impl directory.
 
-import {AmpA4A, DEFAULT_SAFEFRAME_VERSION} from '../../amp-a4a/0.1/amp-a4a';
+import {
+  AmpA4A,
+  RENDERING_TYPE_HEADER,
+  XORIGIN_MODE,
+} from '../../amp-a4a/0.1/amp-a4a';
 import {
   isInManualExperiment,
 } from '../../../ads/google/a4a/traffic-experiments';
 import {
   extractGoogleAdCreativeAndSignature,
   googleAdUrl,
+  truncAndTimeUrl,
+  googleBlockParameters,
+  googlePageParameters,
   isGoogleAdsA4AValidEnvironment,
   AmpAnalyticsConfigDef,
   extractAmpAnalyticsConfig,
+  groupAmpAdsByType,
 } from '../../../ads/google/a4a/utils';
 import {getMultiSizeDimensions} from '../../../ads/google/utils';
 import {
   googleLifecycleReporterFactory,
   setGoogleLifecycleVarsFromHeaders,
 } from '../../../ads/google/a4a/google-data-reporter';
+import {
+  fetchLineDelimitedChunks,
+} from '../../../ads/google/a4a/line-delimited-response-handler';
 import {stringHash32} from '../../../src/crypto';
 import {removeElement} from '../../../src/dom';
-import {dev} from '../../../src/log';
-import {extensionsFor} from '../../../src/services';
+import {dev, user} from '../../../src/log';
+import {getMode} from '../../../src/mode';
+import {extensionsFor, xhrFor} from '../../../src/services';
 import {isExperimentOn} from '../../../src/experiments';
 import {domFingerprintPlain} from '../../../src/utils/dom-fingerprint';
 import {insertAnalyticsElement} from '../../../src/analytics';
 import {setStyles} from '../../../src/style';
+import {utf8Encode} from '../../../src/utils/bytes';
+import {isCancellation} from '../../../src/error';
+
+/** @type {string} */
+const TAG = 'amp-ad-network-doubleclick-impl';
 
 /** @const {string} */
 const DOUBLECLICK_BASE_URL =
     'https://securepubads.g.doubleclick.net/gampad/ads';
 
+/** @private @const {!Object<string,string>} */
+const PAGE_LEVEL_PARAMS_ = {
+  'gdfp_req': '1',
+  'sfv': 'A',
+  'u_sd': window.devicePixelRatio,
+};
+
+/** @private @const {string} */
+const TFCD_ = 'tagForChildDirectedTreatment';
+
+/** @private {?Promise} */
+let SraRequests = null;
```

Comment:
This is actually a global instance shared across all doubleclick instances.  Still needs to be lowercase?